wifiphisher/common/deauth.py

```diff
@@ -42,39 +44,54 @@ def __init__(self, ap_bssid, jamming_interface):
         self._ap_bssid = ap_bssid
         self._should_continue = True
         self._jamming_interface = jamming_interface
+        self._ap_iface_mac = ap_iface_mac
         self._non_client_addresses = constants.NON_CLIENT_ADDRESSES
-        self._send_deauth_packets_thread = threading.Thread(\
-                target = self._send_deauthentication_packets)
-        self._find_client_thread = threading.Thread(\
-                target=self._find_clients)
+        self._deauth_thread = None
+        self._find_client_thread = None
 
         # create a socket for sending packets
         self._socket = linux.L2Socket(iface=self._jamming_interface)
 
         # craft and add deauthentication packet to broadcast address
-        self._craft_and_add_packet(self._ap_bssid, constants.WIFI_BROADCAST)
+        # add broadcast deauth when --essid not being used
+        if not self._is_frenzy_deauth():
```

Comment:
For example if we did these operations in another function, I can construct the `NetworkManager` directly, and for some function related to HW, I can simply patch these functions. But for now, I can just mock the NetworkManager and have to define the functions even if not related to hardware specific for NetworkManager. Is this also one of the reason you want to refactor NetworkManager and Adapter class?